tools/kind-d8.sh

```diff
@@ -0,0 +1,502 @@
+
+    if [ "$?" -ne "0" ]; then
+      echo "Unable to download kind."
+      exit 1
+    fi
+
+    install -m 0755 kind "${KIND_INSTALL_DIRECTORY}"/kind
+
+    if [ "$?" -ne "0" ]; then
+      echo "Insufficient permissions to install kind. Trying again with sudo..."
+      sudo install -m 0755 kind "${KIND_INSTALL_DIRECTORY}"/kind
+      if [ "$?" -ne "0" ]; then
+        echo "Unable to install kind. Check installation path and permissions."
+        exit 1
+      fi
+    fi
+
+    KIND_PATH=${KIND_INSTALL_DIRECTORY}/kind
+  fi
+}
+
+preinstall_checks() {
+  local answer=
+  local cluster_exist=true
+
+  read -rp "Specify kind cluster name [$KIND_CLUSTER_NAME]: " answer
+  if [[ -n "$answer" ]]; then
+    KIND_CLUSTER_NAME=$answer
+  fi
+
+  while [[ "$cluster_exist" == "true" ]]; do
+
+    # Check if a kind cluster with the name `d8` exist
+    ${KIND_PATH} get clusters | grep -q "^${KIND_CLUSTER_NAME}$" &>/dev/null
+
+    if [ "$?" -eq "0" ]; then
+      cluster_exist=true
+    else
+      cluster_exist=false
+    fi
+
+    while [[ ("$answer" != "y") && ("$cluster_exist" == "true") ]]; do
+      read -rp "A kind cluster named '$KIND_CLUSTER_NAME' already exists. Delete it? y/[n]: " answer
+      if [[ ("$answer" == "n") || (-z "$answer") ]]; then
+        echo -e "Please delete the cluster '${KIND_CLUSTER_NAME}' manually. You can use the following command:\n\t${KIND_PATH} delete cluster --name ${KIND_CLUSTER_NAME}."
+        exit 1
+      fi
+
+      if [[ "$answer" != "y" ]]; then
+        echo "Please type 'y' to delete cluster '${KIND_CLUSTER_NAME}' or 'n' to abort the installation."
+      fi
+    done
+
+    if [[ "$cluster_exist" == "true" ]]; then
+      ${KIND_PATH} delete cluster --name "${KIND_CLUSTER_NAME}"
+      sleep 3
+    fi
+  done
+}
+configs_create() {
+  echo "Creating ${CONFIG_DIR} directory..."
+  mkdir -p ${CONFIG_DIR}
+
+  echo "Creating kind config file (${CONFIG_DIR}/kind.cfg)..."
+  cat <<EOF >${CONFIG_DIR}/kind.cfg
+apiVersion: kind.x-k8s.io/v1alpha4
+kind: Cluster
+nodes:
+- role: control-plane
+  extraPortMappings:
+  - containerPort: 80
+    hostPort: 80
+    listenAddress: "127.0.0.1"
+    protocol: TCP
+  - containerPort: 443
+    hostPort: 443
+    listenAddress: "127.0.0.1"
+    protocol: TCP
+EOF
+
+  echo "Creating Deckhouse installation config file (${CONFIG_DIR}/config.yml)..."
+  cat <<EOF >${CONFIG_DIR}/config.yml
+apiVersion: deckhouse.io/v1
+kind: InitConfiguration
+deckhouse:
+  releaseChannel: Stable
+  bundle: Minimal
+  configOverrides:
+    global:
+      modules:
+        publicDomainTemplate: "%s-127-0-0-1.nip.io"
+        https:
+          mode: Disabled
+    operatorPrometheusCrdEnabled: true
+    operatorPrometheusEnabled: true
+    prometheusCrdEnabled: true
+    prometheusEnabled: true
+    prometheus:
+      longtermRetentionDays: 0
+    monitoringKubernetesControlPlaneEnabled: true
```

Comment:
I think we have a problem with kind installation. The Grafana is pretty empty right after the installation. I did not manage to find a working dashboard in the first five clicks. Maybe we need to enable a monitoring-kubernetes module also.

Reply:
Ok. I've enabled the `monitoring-kubernetes` module. I see that there are dashboards with data.